### ISSUE TITLE
Fastapi: multiple http receive and send spans for single request 

### ISSUE THREAD

**Issue by Nan2018** [open-telemetry/opentelemetry-python-contrib#635]
I am using `opentelemetry-instrumentation-fastapi`. Everything works great except I get many small spans of `http receive` and `http send` for a single request. (see the screenshot below) Is there a way to combine these spans?
![image](https://user-images.githubusercontent.com/37547264/131183098-f3de7d3c-eb9c-4935-b8f4-986d948cea62.png)

**Comment by adriangb** [comment]
This is per design I think: there are spans for each time data is received and sent (i.e. things are going over the wire). I think this is important because it would let you detect if for example the issue is a slow client.

**Comment by niehongxu116** [comment]
I also have a similar issue with a POST endpoint that uses JSON transmission. In the trace, there are multiple HTTP receive events. I suspect this might be because the request body is too large and exceeds the buffer size, or is it caused by something else?

```
uvicorn == 0.31.1
fastApi == 0.115.6
opentelemetry-instrumentation-fastapi == 0.49b2
```

![Image](https://github.com/user-attachments/assets/21b3442b-33e3-4329-8cdd-eb66304e3c16)

**Comment by SuperSecureHuman** [comment]
So in this case, is the workaround able to preserve the inputs and the outputs of this route.. I have a similar issue working with FastAPI streaming responses..

At the moment, I would not want to trace the chunks (events) rather the input and output and overall latency of the route (from request to stream end)

<img width="1736" alt="Image" src="https://github.com/user-attachments/assets/5c6351ac-2fe4-4d2e-8bed-079ddcba2206" />

**Comment by srprash** [comment]
> Exact same as above on POST of a large body. 10,000 spans generated.
> 
> Seems like a duplicate of [this](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/831) which contains a workaround and a link to the fix [here](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2802/files).
> 
> TLDR - Add exclude_spans e.g., `FastAPIInstrumentor.instrument_app(app, exclude_spans=["send", "receive"])`

Is there a way to exclude spans when using OpenTelemetry Python zero-code instrumentation?

**Comment by ocelotl** [comment]
I asked Claude to review this issue and Claude suggests it be closed with this explanation:

> Fixed by merged PR #2802 (`exclude_spans=["send","receive"]`); dup of closed #831.
>
> Proof in `main`: [`fastapi/__init__.py#L244`](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/94e851dfe333d47fdb52edb82635330668af284e/instrumentation/opentelemetry-instrumentation-fastapi/src/opentelemetry/instrumentation/fastapi/__init__.py#L244-L244)

To guard against a mistaken suggestion, Claude then independently re-verified this conclusion:

> Verified: `fastapi/__init__.py` and the ASGI middleware implement `exclude_spans` with receive/send filtering, so the extra spans can be suppressed.

Claude evaluated this issue with high confidence that it can be closed.

Claude has classified this issue as already resolved (the requested fix or feature has already shipped).

The intention of this comment is to make it easier for a human being to review this issue before making a decision.
